test/e2e/adminapi_redeployvm.go

```diff
@@ -108,25 +109,30 @@ func getNodeUptime(g Gomega, ctx context.Context, node string) (time.Time, error
 		return time.Time{}, err
 	}
 
-	// Defer Delete
 	defer func() {
-		By("deleting uptime pod")
-		err := clients.Kubernetes.CoreV1().Pods(namespace).Delete(ctx, name, metav1.DeleteOptions{})
-		if err != nil {
-			log.Error("Could not delete test Pod")
-		}
+		By("deleting the uptime pod via Kubernetes API")
+		deleteCall := clients.Kubernetes.CoreV1().Pods(namespace).Delete
+		DeleteK8sObjectWithRetry(ctx, deleteCall, podName, metav1.DeleteOptions{})
+
+		// To avoid flakes, we need it to be completely deleted before we can use it again
+		// in a separate run or in a separate It block
+		By("waiting for uptime pod to be deleted")
```

Comment:
I'll leave it as is for now since this comment is getting nuked in a subsequent PR that I'll open after this: https://github.com/mociarain/ARO-RP/pull/1.